### PR TITLE
[DF] Add more tests for Vary

### PR DESCRIPTION
**Pull request by eguiraud**
With many thanks to @ikabadzhov who did the bulk of the work.

### CODE REVIEW

File: tree/dataframe/test/dataframe_vary.cxx [hunk 1]
```diff
@@ -654,19 +713,623 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
...
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Stats<int>("x");
+   EXPECT_THROW(VariationsFor(h), std::runtime_error); // FIXME not (yet) implemented
```
Comment (ikabadzhov) [inline tree/dataframe/test/dataframe_vary.cxx:1248]: I didn't expect to manage this one, so I left it as a runtime error. I am now adapting the test to verify the expected behaviour.

File: tree/dataframe/test/dataframe_vary.cxx [hunk 2]
```diff
@@ -125,6 +136,33 @@ TEST(RDFVary, VariationsForDoesNotTriggerRun)
    EXPECT_EQ(df.GetNRuns(), 0);
 }
 
+TEST(RDFVary, InvalidQueries)
+{
+   auto df = ROOT::RDataFrame(10).Define("x", [] { return 1.f; });
+   // x is float, variation expression cannot return RVec<int>, must be RVec<float>
+   EXPECT_THROW(df.Vary("x", SimpleVariation, {}, 2), std::runtime_error);
```
Comment (ikabadzhov) [inline tree/dataframe/test/dataframe_vary.cxx:143]: Here, and on some other places I *think* it is a good idea to check what is the exact message thrown. I did not do that previously, as we were continuously changing few of the error messages.

I am taking care of it now.

File: tree/dataframe/test/dataframe_vary.cxx [hunk 3]
```diff
@@ -162,6 +200,27 @@ TEST(RDFVary, VaryDefinePerSample)
...
+   ROOT::RDataFrame df(1);
+   auto c = df.Define("x", [] { return 0; })
+               .Vary(
+                  "x",
+                  [] {
+                     return ROOT::RVecI{0, 0};
+                  },
+                  {}, 2)
+               .Count();
+   auto cs = ROOT::RDF::Experimental::VariationsFor(c);
```
Comment (ikabadzhov) [inline tree/dataframe/test/dataframe_vary.cxx:214]: I realize that `ROOT::RDF::Experimental::` is redundant. Removing it now.

File: tree/dataframe/test/dataframe_vary.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/test/dataframe_vary.cxx:1248]: you added a test with the PR that fixes Vary+Stats. do we still need this test case?

**Comment by eguiraud** [comment]
@phsft-bot build

**Comment by eguiraud** [comment]
The test failure on Windows is real, @ikabadzhov can you please investigate with @bellenot 's help when you are back?

**Comment by eguiraud** [comment]
@phsft-bot build please

**Review by ikabadzhov — APPROVED** [review]
Looks good. This was a lengthy one!

**Pull request merged**